### ISSUE TITLE
Unecessarily public item

### ISSUE THREAD

**Issue by Kmeakin** [rust-lang/rust-clippy#11784]
### What it does

Lint items that are exported as `pub`, but are not used outside of the module they re defined in. 

### Advantage

Ensure code is not more public than it needs to be, enforcing abstraction boundaries

### Drawbacks

High possibility of false positives. Should be in the `pedantic` category, and have some heuristics for ignoring items that are intended to be exported from the crate to other crates (eg don't lint on items exported from `lib.rs` or `prelude.rs`)

### Example

```rust
mod foo {
    pub fn f() {}

    pub fn g() { 
        f()
    }
}
```

Could be written as:

```rust
mod foo {
    fn f() {}

    pub fn g() { 
        f()
    }
}
```
